test/boost/commitlog_test.cc

```diff
@@ -352,15 +352,21 @@ SEASTAR_TEST_CASE(test_commitlog_delete_when_over_disk_limit) {
 SEASTAR_TEST_CASE(test_commitlog_reader){
     static auto count_mutations_in_segment = [] (sstring path) -> future<size_t> {
         size_t count = 0;
-        co_await db::commitlog::read_log_file(path, db::commitlog::descriptor::FILENAME_PREFIX, [&count](db::commitlog::buffer_and_replay_position buf_rp) -> future<> {
-            auto&& [buf, rp] = buf_rp;
-            auto linearization_buffer = bytes_ostream();
-            auto in = buf.get_istream();
-            auto str = to_sstring_view(in.read_bytes_view(buf.size_bytes(), linearization_buffer));
-            BOOST_CHECK_EQUAL(str, "hej bubba cow");
-            count++;
-            co_return;
-        });
+        try {
+            co_await db::commitlog::read_log_file(path, db::commitlog::descriptor::FILENAME_PREFIX, [&count](db::commitlog::buffer_and_replay_position buf_rp) -> future<> {
+                auto&& [buf, rp] = buf_rp;
+                auto linearization_buffer = bytes_ostream();
+                auto in = buf.get_istream();
+                auto str = to_sstring_view(in.read_bytes_view(buf.size_bytes(), linearization_buffer));
+                BOOST_CHECK_EQUAL(str, "hej bubba cow");
+                count++;
+                co_return;
+            });
+        } catch (commitlog::segment_truncation&) {
+            // ok. this does not ensure to have fully synced segments
+            // before reading them in all code paths. We can end up
+            // hitting (premature) eof or even half-written page.
+        }
```

Comment:
So, if read_log_path() is changed to always throw segment_truncation, the test will pass.

What are we testing here? We need to check that we read the data that we wrote, not that we obey some protocol.

Reply:
No, if we get truncation, we accept it as a segment stop marker. Note the data check in the replay callback + the "count" var being incremented. The code after this will verify this var has the correct value (i,e. replayed all entries we expected.

Reply:
ok